discovery/main.go

```diff
@@ -0,0 +1,28 @@
+package discovery
+
+import (
+	"net"
+)
+
+type Endpoint struct {
+	Name string
+	Port uint16
+}
+
+func DiscoverEndpoints(serviceDNS string) ([]Endpoint, error) {
```

Comment:
personally, I'd do something like this
```
func DiscoverEndpoints(serviceName string) ([]Endpoint, error) {
  _, remotes, err := net.LookupSRV("", "", serviceName) 
  if err != nil {
    return nil, err
  }

  endpoints := []Endpoint{}
  for _, n := range remotes {
    endpoints = append(...)
  }
  return endpoints, nil
}
```

a little different, mostly around some spacing and where I'd declare endpoints. That is mostly style though. 

I am not sure if we need the distinction though. I mean the remote structure is pretty close to the Endpoint in general. I don't think we are getting much by doing this in commons really.

Reply:
Initially I wanted to add more methods to ```Endpoint```, but I guess it is worthless right now